keras_nlp/src/models/vit_det/vit_det_backbone.py

```diff
@@ -0,0 +1,210 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import keras
+from keras import ops
+
+from keras_nlp.src.api_export import keras_nlp_export
+from keras_nlp.src.models.backbone import Backbone
+from keras_nlp.src.models.vit_det.vit_layers import AddPositionalEmbedding
+from keras_nlp.src.models.vit_det.vit_layers import ViTDetPatchingAndEmbedding
+from keras_nlp.src.models.vit_det.vit_layers import WindowedTransformerEncoder
+
+
+@keras_nlp_export("keras_nlp.models.ViTDetBackbone")
```

Comment:
We avoid the weird CaPITalIZatiON in model names for usability. Let's do `VitDet` instead?

Reply:
its ViT - because of Vision Transformer. It is ViT almost everywhere https://huggingface.co/docs/transformers/en/model_doc/vitdet
Fine to change it to VitDet, just double checking.

Reply:
Yeah, I see what you mean. We've been doing this -> https://github.com/keras-team/keras-nlp/blob/master/STYLE_GUIDE.md#naming-of-layers-and-models

This is because trying to follow all the capitalization on the nlp side would be a nightmare. RoBERTa, GPTNeoX, ALBERT. There's case all over the place, with unclear patterns.

So using ViT here seems slightly inconsistent, but maybe ok given how common that spelling is.

Reply:
I see! In that case we can keep it consistent with the rest of the library. I will change this to `VitDetBackbone`